lib/js-yaml/dumper.js

```diff
@@ -272,7 +265,7 @@ function chooseScalarStyle(string, singleLineOnly, indentPerLevel, lineWidth, te
       if (!isPrintable(char)) {
         return STYLE_DOUBLE;
       }
-      plain = plain && isPlainSafe(char);
+      plain = plain && (isPlainSafe(char) && !isWhitespace(string.charCodeAt(i + 1)));
```

Comment:
`isWhitespace(string.charCodeAt(i + 1))` missed length check, to avoid out of bounds access at [i + 1]. Such thing will trash JIT optimisations.

Reply:
Should it check next or prev char?

`{ test: 'C #123' }` - make sure this will be quoted.